src/mentorship/ml/models/common.py

```diff
@@ -1,39 +1,94 @@
-from sklearn.base import clone
+from numpy.lib.stride_tricks import sliding_window_view
+from sklearn.base import clone, BaseEstimator
+import numpy as np
+import pandas as pd
 
 
-class SplitPipeline:
-    def __init__(self, base_pipeline):
-        self.base_pipeline = base_pipeline
+class RecursiveTSEstimator(BaseEstimator):
+    def __init__(self, split_key, target_col, base_pipelines=None, zero_categories=None):
```

Comment:
`zero_categories` is probably specific to your current task. I would suggest you removing it from the generic `RecursiveTSEstimator`, but make a child class for your task with this parameter.